modules/statistics/server/controllers/statistics.server.controller.js

```diff
@@ -187,6 +188,51 @@ exports.getPushRegistrationCount = function (callback) {
   });
 };
 
+/**
+ * Generate statistics based on the user last seen attribute
+ */
+exports.getLastSeenStatistic = function (type, callback) {
+  var now = moment();
+  var query = { };
+
+  switch (type) {
+    case 'never':
+      query.seen = { $exists: false };
+      break;
+    case 'today':
+      query.seen = { '$gte': now.startOf('day') };
+      break;
+    case 'week':
+      query.seen = { '$gte': now.startOf('week') };
+      break;
+    case 'thisMonth':
+      query.seen = { '$gte': now.startOf('month') };
+      break;
+    case 'sinceAMonthAgo':
+      var thisMonth = now.startOf('month');
+      query.seen = { '$gte': thisMonth.subtract(1, 'months') };
+      break;
+    case 'sinceSixMonthsAgo':
+      var thisMonth = now.startOf('month');
+      query.seen = { '$gte': thisMonth.subtract(2, 'months') };
+      break;
+    case 'sinceAYearAgo':
+      var thisMonth = now.startOf('month');
+      query.seen = { '$gte': thisMonth.subtract(12, 'months') };
+      break;
+    default:
```

Comment:
Stats for the current ongoing month/week/day might not give us any real insight as a snapshot stat. Sliding window of "past X days" would be more useful metric and we can work on improving.

How about these sliding windows: 
- past 48h,
- past 7 days,
- past 14 days,
- past 30 days,
- past 6 months,
- past 365 days